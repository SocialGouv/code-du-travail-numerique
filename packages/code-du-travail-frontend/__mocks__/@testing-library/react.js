/* eslint-disable import/export */
import { theme } from "@socialgouv/cdtn-ui";
import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
