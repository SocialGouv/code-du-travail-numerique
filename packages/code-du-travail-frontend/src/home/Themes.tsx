import { getRouteBySource, SOURCES } from "@socialgouv/cdtn-types";
import {
  Button,
  Container,
  GridCell,
  icons,
  PageTitle,
  RootGrid,
  Section,
  theme,
} from "@socialgouv/cdtn-ui";
import Link from "next/link";
import React from "react";
import styled from "styled-components";
import { LinkedTile } from "../common/tiles/LinkedTile";

type Props = {
  themes: Array<{
    icon: string;
    slug: string;
    title: string;
  }>;
};

export const Themes = ({ themes = [] }: Props) => {
  const enrichedThemes = themes.map(({ icon, slug, title }) => {
    return (
      <GridCell key={slug}>
        <LinkedTile
          title={title}
          icon={icons[icon]}
          titleTagType="h3"
          key={slug}
          href={`/${getRouteBySource(SOURCES.THEMES)}/${slug}`}
        />
      </GridCell>
    );
  });

  const [t1, t2, ...otherThemes] = enrichedThemes;

  const titleCell = (
    <DesktopOnlyGridCell key="title cell">
      <ShiftedPageTitleWrapper>
        <PageTitle
          as="h2"
          stripe="left"
          subtitle="Retrouvez tous nos contenus organisés par thèmes"
        >
          Thèmes
        </PageTitle>
      </ShiftedPageTitleWrapper>
    </DesktopOnlyGridCell>
  );

  const ghostCell = <DesktopOnlyGridCell key="ghost cell" />;

  const illustrationCell = (
    <DesktopOnlyGridCell key="icon">
      <StyledThemeIcon />
    </DesktopOnlyGridCell>
  );

  return (
    <Section large decorated variant="light">
      <RelativeContainer>
        <DesktopBackground>
          <StyledShadeIcon />
        </DesktopBackground>
        <MobileTitleWrapper>
          <PageTitle
            as="h2"
            subtitle="Retrouvez tous nos contenus organisés par thèmes"
          >
            Thèmes
          </PageTitle>
        </MobileTitleWrapper>
        <RootGrid>
          {[titleCell, ghostCell, t1, t2, illustrationCell].concat(otherThemes)}
        </RootGrid>
        <Link href="/themes" passHref legacyBehavior>
          <Button variant="primary" as="a">
            Voir tous les thèmes <StyledArrowRight />
          </Button>
        </Link>
      </RelativeContainer>
    </Section>
  );
};

const { breakpoints, spacings } = theme;

const DesktopOnlyGridCell = styled(GridCell)`
  display: ${({ theme }) => (theme.noColors ? "none" : "flex")};
  align-items: center;
  justify-content: center;
  max-height: 18rem;
  overflow: visible;
  @media (max-width: ${breakpoints.desktop}) {
    display: none;
  }
`;

const MobileTitleWrapper = styled.div`
  display: none;
  @media (max-width: ${breakpoints.desktop}) {
    display: block;
  }
`;

const RelativeContainer = styled(Container)`
  position: relative;
  text-align: center;
`;

const DesktopBackground = styled.div`
  position: absolute;
  top: 13rem;
  left: 3rem;
  display: ${({ theme }) => (theme.noColors ? "none" : "block")};
  width: 81rem;
  height: 52rem;
  @media (max-width: ${breakpoints.desktop}) {
    display: none;
  }
`;

const StyledShadeIcon = styled(icons.Shade)`
  width: 100%;
  color: ${({ theme }) => theme.secondary};
`;

const ShiftedPageTitleWrapper = styled.div`
  width: 100%;
  transform: translateX(20%);
`;

const StyledThemeIcon = styled(icons.WorkersTheme)`
  flex: 1 1 auto;
  height: 115%;
  overflow: visible;
`;

const StyledArrowRight = styled(icons.DirectionRight)`
  width: 2.8rem;
  height: 1.5rem;
  margin-left: ${spacings.base};
`;
