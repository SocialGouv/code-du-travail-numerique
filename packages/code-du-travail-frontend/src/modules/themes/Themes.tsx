import { fr } from "@codegouvfr/react-dsfr";
import React from "react";
import { ThemeTile } from "./ThemeTile";
import { ThemeElasticDocument } from "@socialgouv/cdtn-types/build/elastic/theme";
import { css } from "@styled-system/css";
import { ContainerList } from "../layout/ContainerList";

type Props = {
  themes: Pick<ThemeElasticDocument, "title" | "icon" | "slug" | "children">[];
};

export const Themes = ({ themes }: Props) => (
  <ContainerList title="Thèmes">
    <h1 id="themes" className={fr.cx("fr-mt-0", "fr-mb-6w")}>
      Thèmes
    </h1>

    <p className={fr.cx("fr-text--md")}>
      Découvrez l’intégralité de nos contenus organisés par grands thèmes
    </p>

    <div className={fr.cx("fr-mt-6w")}>
      <ul
        className={fr.cx(
          "fr-grid-row",
          "fr-grid-row--gutters",
          "fr-grid-row--left"
        )}
      >
        {themes.map((item, index) => (
          <li
            key={`${index}${JSON.stringify(item)}`}
            className={`${fr.cx(
              "fr-col-12",
              "fr-col-sm-6",
              "fr-col-md-4",
              "fr-col-lg-3",
              "fr-mb-3v"
            )} ${li}`}
          >
            <ThemeTile
              iconName={item.icon!!}
              title={item.title}
              link={`/themes/${item.slug}`}
              subThemes={item.children.map((child) => child.label)}
            />
          </li>
        ))}
      </ul>
    </div>
  </ContainerList>
);

const li = css({
  listStyle: "none!",
});
