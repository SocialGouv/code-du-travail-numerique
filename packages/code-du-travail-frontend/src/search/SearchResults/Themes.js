import { getRouteBySource } from "@socialgouv/cdtn-utils";
import { Grid, Title } from "@socialgouv/cdtn-ui";
import React from "react";

import { reportSelectionToMatomo } from "../utils";
import { LinkedTile } from "../../common/tiles/LinkedTile";

export const Themes = ({ items, query }) => (
  <>
    <Title>Les thÃ¨mes suivants peuvent vous intÃ©resser</Title>
    <Grid>
      {items.map(({ slug, title, source, url, algo }) => (
        <LinkedTile
          onClick={() => reportSelectionToMatomo(source, slug, url, algo)}
          title={title}
          striped
          key={slug}
          href={`/${getRouteBySource(source)}/${slug}${
            query ? `?q=${query}` : ""
          }`}
        />
      ))}
    </Grid>
  </>
);
