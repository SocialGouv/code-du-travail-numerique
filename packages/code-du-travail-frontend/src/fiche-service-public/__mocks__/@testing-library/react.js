import React from "react";
import PropTypes from "prop-types";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "@socialgouv/cdtn-ui";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
