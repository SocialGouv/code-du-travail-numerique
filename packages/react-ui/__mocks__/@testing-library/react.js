import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

import { colors } from "../../src/theme";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
