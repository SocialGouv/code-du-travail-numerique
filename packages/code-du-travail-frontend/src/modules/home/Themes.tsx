import Image from "next/image";
import { HomeButton, SectionContainer } from "./Components";
import { fr } from "@codegouvfr/react-dsfr";
import { MatomoHomeEvent, useHomeTracking } from "./tracking";
import { HomeTileItem } from "./queries";
import IllustrationTheme from "./picto/themes.svg";
import { ThemeTile } from "../themes/ThemeTile";

type Props = {
  items: HomeTileItem[];
};

export const Themes = (props: Props) => {
  const { emitHomeClickButtonEvent } = useHomeTracking();
  return (
    <SectionContainer
      sectionId="home-themes"
      title="Thèmes"
      subtitle="Retrouvez tous nos contenus organisés par thèmes"
      footerNode={
        <HomeButton
          buttonLink="/themes"
          buttonText="Voir tous les thèmes"
          onButtonClick={() => {
            emitHomeClickButtonEvent(
              MatomoHomeEvent.CLICK_VOIR_TOUTES_LES_THEMES
            );
          }}
        />
      }
    >
      <>
        <div className={fr.cx("fr-col-4", "fr-hidden", "fr-unhidden-lg")}>
          <Image src={IllustrationTheme} alt="" width="384" height="233" />
        </div>
        {props.items.map((item, index) => (
          <div
            key={`${index}${JSON.stringify(item)}`}
            className={fr.cx(
              "fr-col-12",
              "fr-col-sm-6",
              "fr-col-md-4",
              "fr-col-lg-2"
            )}
          >
            <ThemeTile {...item} />
          </div>
        ))}
      </>
    </SectionContainer>
  );
};
