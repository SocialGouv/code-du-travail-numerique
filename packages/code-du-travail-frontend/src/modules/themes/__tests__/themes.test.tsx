import { getAllByRole, render } from "@testing-library/react";
import { Themes } from "../Themes";

afterEach(() => {
  jest.resetAllMocks();
});

const dataThemes = [
  {
    title: "Theme 1",
    icon: "",
    slug: "theme1",
    children: [
      {
        label: "children 1",
        slug: "children1_1",
      },
      {
        label: "children 2",
        slug: "children1_2",
      },
    ],
  },
  {
    title: "Theme 2",
    icon: "",
    slug: "theme2",
    children: [
      {
        label: "children 3",
        slug: "children2_3",
      },
      {
        label: "children 4",
        slug: "children2_4",
      },
    ],
  },
];

describe("<Themes />", () => {
  it("affiche la liste des thÃ¨mes principaux", () => {
    const { container } = render(<Themes themes={dataThemes} />);
    const themesList = getAllByRole(container, "list")[1];
    const themes = getAllByRole(themesList, "listitem");
    expect(themes).toHaveLength(2);
    expect(getAllByRole(themes[0], "link")).toHaveLength(1);
    expect(getAllByRole(themes[0], "link")[0].getAttribute("href")).toEqual(
      "/themes/theme1"
    );

    expect(container).toMatchSnapshot();
  });
});
