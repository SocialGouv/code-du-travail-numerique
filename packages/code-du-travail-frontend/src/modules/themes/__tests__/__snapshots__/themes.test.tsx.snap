// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Themes /> affiche la liste des thèmes principaux 1`] = `
<div>
  <div
    class="fr-grid-row"
  >
    <nav
      aria-label="vous êtes ici :"
      class="fr-breadcrumb fr-mb-2v"
      id="fr-breadcrumb-«r0»"
      role="navigation"
    >
      <button
        aria-controls="breadcrumb-«r1»"
        aria-expanded="false"
        class="fr-breadcrumb__button"
      >
        Voir le fil d’Ariane
      </button>
      <div
        class="fr-collapse"
        id="breadcrumb-«r1»"
      >
        <ol
          class="fr-breadcrumb__list"
        >
          <li>
            <a
              class="fr-breadcrumb__link"
              href="/"
            >
              Accueil
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="fr-breadcrumb__link"
            >
              Thèmes
            </a>
          </li>
        </ol>
      </div>
    </nav>
    <div
      class="fr-col-12 fr-mb-4w fr-mb-md-12w"
    >
      <h1
        class="fr-mt-0 fr-mb-6w"
        id="themes"
      >
        Thèmes
      </h1>
      <p
        class="fr-text--md"
      >
        Découvrez l’intégralité de nos contenus organisés par grands thèmes
      </p>
      <div
        class="fr-mt-6w"
      >
        <ul
          class="fr-grid-row fr-grid-row--gutters fr-grid-row--left"
        >
          <li
            class="fr-col-12 fr-col-sm-6 fr-col-md-4 fr-col-lg-3 fr-mb-3v li-s_none!"
          >
            <div
              class="fr-tile fr-enlarge-link"
            >
              <p
                class="fr-tile__title"
              >
                <a
                  href="/themes/theme1"
                >
                  Theme 1
                </a>
              </p>
              <p
                class="fr-tile__detail fr-mb-4w jc_center"
              >
                children 1 • children 2
              </p>
              <div
                class="fr-tile__img"
              >
                <img
                  alt=""
                  data-nimg="1"
                  decoding="async"
                  height="80"
                  loading="lazy"
                  src="/static/assets/icons/themes/.svg"
                  style="color: transparent;"
                  width="80"
                />
              </div>
            </div>
          </li>
          <li
            class="fr-col-12 fr-col-sm-6 fr-col-md-4 fr-col-lg-3 fr-mb-3v li-s_none!"
          >
            <div
              class="fr-tile fr-enlarge-link"
            >
              <p
                class="fr-tile__title"
              >
                <a
                  href="/themes/theme2"
                >
                  Theme 2
                </a>
              </p>
              <p
                class="fr-tile__detail fr-mb-4w jc_center"
              >
                children 3 • children 4
              </p>
              <div
                class="fr-tile__img"
              >
                <img
                  alt=""
                  data-nimg="1"
                  decoding="async"
                  height="80"
                  loading="lazy"
                  src="/static/assets/icons/themes/.svg"
                  style="color: transparent;"
                  width="80"
                />
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
